enclone_args/src/load_gex.rs

```diff
@@ -86,29 +76,29 @@ pub fn get_gex_info(ctl: &mut EncloneControl) -> Result<GexInfo, String> {
     let mut h5_data = Vec::<Option<Dataset>>::new();
     let mut h5_indices = Vec::<Option<Dataset>>::new();
     let mut h5_indptr = Vec::<Vec<u32>>::new();
-    if ctl.gen_opt.h5 {
-        let gex_outs = &ctl.origin_info.gex_path;
```

Comment:
Could there be any inter-dependency between these two? Maybe some validation on the enclone side that `ctl.origin_info.gex_path` is present and is of length `ctl.origin_info.dataset_path.len()` if `ctl.gen_opt.h5` is true. Wondering if we can get index out of bounds for `gex_outs[i]`

Reply:
In this refactoring process I've been trying to be very careful to make sure that I've confirmed that a conditional is constrained to exactly one path before removing it.